staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -129,7 +129,7 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.R
 	clientCertificateExpirationHistogram.Observe(remaining.Seconds())
 	chains, err := req.TLS.PeerCertificates[0].Verify(optsCopy)
 	if err != nil {
-		return nil, false, err
+		return nil, false, fmt.Errorf("verifying cert for %s/%s failed: %w", req.TLS.PeerCertificates[0].Subject.CommonName, req.TLS.PeerCertificates[0].Subject.Organization, err)
```

Comment:
I think it might make sense to go as far as to printing the subject and the issuer

Reply:
done